src/main/java/uk/gov/companieshouse/service/dissolution/DissolutionCreator.java

```diff
@@ -29,10 +30,10 @@ public DissolutionCreator(
         this.responseMapper = responseMapper;
     }
 
-    public DissolutionCreateResponse create(DissolutionCreateRequest body, String companyNumber, String userId, String ip, String email) {
+    public DissolutionCreateResponse create(DissolutionCreateRequest body, CompanyProfileApi companyProfileApi, String userId, String ip, String email) {
         final String reference = referenceGenerator.generateApplicationReference();
 
-        final Dissolution dissolution = requestMapper.mapToDissolution(body, companyNumber, userId, email, ip, reference);
+        final Dissolution dissolution = requestMapper.mapToDissolution(body, companyProfileApi.getCompanyNumber(), companyProfileApi.getCompanyName(), userId, email, ip, reference);
```

Comment:
could you just pass the companyProfileApi through to the mapper and let it extract the fields rather than this service?